masonry/src/properties/types/as_unit.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::properties::types::Length;
+
+// NOTE ON OVERFLOW:
+// Casting a u64 to f64 can never overflow. In some cases, we may lose precision.
+// This is mostly fine, given that lengths large enough to overflow an f64 will be extremely rare.
+// See reference for details on casts:
+// https://doc.rust-lang.org/reference/expressions/operator-expr.html#r-expr.as.numeric.int-as-float
+
+/// Utility trait for wrapping numbers in logical units.
+pub trait AsUnit {
+    /// Create a length, in logical pixels.
+    ///
+    /// # Panics
+    ///
+    /// Panics if value is negative, infinite.
+    fn px(self) -> Length;
```

Comment:
Given that Jetpack compose uses Px for physical pixels (see e.g. <https://developer.android.com/reference/kotlin/androidx/compose/ui/unit/Density#(androidx.compose.ui.unit.Dp).toPx()>), I'm not convinced that this is the name we should use.